src/ui/components/VPNExpandableAppList.qml

```diff
@@ -10,140 +10,118 @@ import QtGraphicalEffects 1.15
+                Layout.rightMargin: defaultMargin - 5
+                source: "../resources/arrow-toggle.svg"
+                transformOrigin: Image.Center
+                smooth: true
+                opacity: applist.count > 0 ?1:0;
+                rotation: isListVisible ? 0 :-90
+            }
+            ColumnLayout {
                 Layout.alignment: Qt.AlignLeft
                 Layout.fillWidth: true
-
-                VPNInterLabel {
-                    id: label
-                    text: header
-                    Accessible.role: Accessible.Heading
-                    color: Theme.fontColorDark
-                    horizontalAlignment: Text.AlignLeft
+                spacing: 0
+
+                RowLayout {
+                    Layout.alignment: Qt.AlignLeft
+                    Layout.fillWidth: true
+
+                    VPNInterLabel {
+                        id: label
+                        text: header
+                        Accessible.role: Accessible.Heading
+                        color: Theme.fontColorDark
+                        horizontalAlignment: Text.AlignLeft
+                    }
+
+                    VPNTextBlock{
+                        visible: !isListVisible || !isEnabled
+                        text: " (%0)".arg(applist.count)
+                    }
                 }
 
-                VPNTextBlock{
-                    visible: !listVisible || applist.count == 0 || !isEnabled
-                    text: " (%0)".arg(applist.count)
+                VPNTextBlock {
+                    id: enabledAppSubtext
+                    text: description // "These apps will (not) use vpn.."
+                    Layout.fillWidth: true
                 }
             }
-
-            VPNTextBlock {
-                id: enabledAppSubtext
-                text: description
-                Layout.fillWidth: true
+            VPNLinkButton{
+                Layout.alignment: Qt.AlignRight
+                // (Un)protect-All Button
+                labelText: actionText
+                onClicked: {
+                    onAction();
+                    isListVisible=false
+                }
+                enabled: isActionEnabled
+                opacity: isActionEnabled ? 1 : 0
             }
         }
-        VPNLinkButton{
-            Layout.alignment: Qt.AlignRight
-            // (Un)protect-All Button
-            labelText: actionText
-            onClicked: {onAction()}
-            enabled: isEnabled
-            opacity: isEnabled ? 1: 0.5
-        }
     }
 
     VPNList {
         id: applist
         model: listModel
-
-        anchors.top: appRowHeader.bottom
-        anchors.topMargin: count > 0 ? 16 : 0
-        anchors.left: appRowHeader.left
-        width: appRow.width
+        width: appListContainer.width
         height: contentItem.childrenRect.height
         spacing: 26
         listName: header
 
-        visible: listVisible && count > 0
+        Layout.preferredHeight: height
```

Comment:
More QML debugger console appeasement: Let's change this to `Layout.preferredHeight: contentItem.childrenRect.height` and then remove line 102